dbus/src/message/matchrule.rs

```diff
@@ -167,4 +167,300 @@ impl<'a> MatchRule<'a> {
         self.msg_type = Some(ty);
         self
     }
+
+    #[cfg(feature = "parser")]
+    /// Tries parsing a MatchRule from a String. Please note however that not all features supported
+    /// by DBus are supported by dbus-rs (yet). args and destinations are not supported yet.
+    pub fn parse(text: &'a str) -> Result<Self, parser::Error> {
+        parser::Parser::new(text)?.parse()
+    }
+}
+
+
+#[cfg(feature = "parser")]
+pub mod parser {
+    // Our grammar:
+    // rules: rule (, rule)*
+    // rule: sender | type | interface | member | path | path_namespace | destination | arg | arg_path
+    // bool : 'true' | 'false'
+    // type: "type" "=" message_type
+    // message_type: "'signal'" | "'method_call'" | "'method_return'" | "'error'"
+    // sender: "sender" "=" string
+    // interface: "interface" "=" string
+    // member: "member" "=" string
+    // path: "path" "=" string
+    // path_namespace: "path_namespace" "=" string
+    // destination: "destination" "=" string
+    // arg: "arg" 0-63 "=" string
+    // arg_path: "arg" 0-63 "path" "=" string
+    // eavesdrop: "eavesdrop" "=" bool
+
+    use std::error::Error as stdError;
+    use arrayvec::ArrayVec;
+    use std::fmt::{Debug, Formatter, Display};
+    use crate::message::MatchRule;
+    use crate::MessageType;
+    use std::convert::TryFrom;
+    use crate::arg::RefArg;
+
+    const MAX_RULE_TOKENS: usize = 16;
+
```

Comment:
There is no word about maximum rule tokens in the D-Bus specification as I can see...? Where does this restriction come from?

Reply:
I took it from the DBus source code: https://github.com/freedesktop/dbus/blob/ef55a3db0d8f17848f8a579092fb05900cc076f5/bus/signals.c#L720

Reply:
Hmm, and above that was a FIXME saying:

 >  * FIXME this is sort of busted now with arg matching, but we let
 > * you match on up to 10 args for now

So they're basically saying that the limit of 16 should be lifted. Which means we should probably not have such an upper limit.

Reply:
For us to get anywhere near the 16 we'd have to implement arg matching first :P But I see where you're coming from. Perhaps we should figure out a way to implement arg matching first.